lib/matplotlib/tests/test_colors.py

```diff
@@ -1380,3 +1380,56 @@ def test_norm_deepcopy():
     assert isinstance(norm2._scale, mscale.LinearScale)
     assert norm2.vmin == norm.vmin
     assert norm2._scale is not norm._scale
+
+
+def test_norm_callback():
+    norm = mcolors.Normalize()
+    x = 0
+
+    def increment():
+        nonlocal x
+        x += 1
+
+    norm.callbacksNorm.connect('changed', increment)
+    # Haven't updated anything, so x should be 0
+    assert x == 0
+
+    # Now change vmin and vmax to test callbacks
+    norm.vmin = 1
+    assert x == 1
+    norm.vmax = 5
+    assert x == 2
+    # callback shouldn't be called if setting to the same value
+    norm.vmin = 1
+    assert x == 2
+    norm.vmax = 5
+    assert x == 2
+
+
+def test_scalarmappable_norm_update():
+    norm = mcolors.Normalize()
+    sm = matplotlib.cm.ScalarMappable(norm=norm, cmap='plasma')
+    # sm doesn't have a stale attribute at first, set it to False
+    sm.stale = False
+    # The mappable should be stale after updating vmin/vmax
+    norm.vmin = 5
+    assert sm.stale
+    sm.stale = False
+    norm.vmax = 5
+    assert sm.stale
+    sm.stale = False
+    norm.clip = True
+    assert sm.stale
+    # change to the CenteredNorm and TwoSlopeNorm to test those
+    # Also make sure that updating the norm directly and with
+    # set_norm both update the Norm callback
+    norm = mcolors.CenteredNorm()
+    sm.norm = norm
+    sm.stale = False
+    norm.vcenter = 1
+    assert sm.stale
+    norm = mcolors.TwoSlopeNorm(vcenter=0, vmin=-1, vmax=1)
+    sm.set_norm(norm)
+    sm.stale = False
+    norm.vcenter = 1
+    assert sm.stale
```

Comment:
Are these asserts enough? I guess it would be nice to know that the `stale` does something to the `sm`.  Maybe a figure equivalency test that tests setting the norm directly versus changing the norm after creation and proving that the two figures are the same?  It would be nice to see that the whole chain works.  See `check_figures_equal` decorators for a way to do this w/o adding extra figures to the repo.

Reply:
BTW, sorry this languished.  Feel free to ping me and I'l try and shepherd through...

Reply:
Somewhat the opposite - if I give something a hard time, but realize I was wrong, I feel some compunction to help correct my mistake.

For the test, I think using the reference figure with the norm set initially, and the compare figure with the norm modified after the fact to match the reference figure should work?  i.e basically what you have here, but a reference figure that starts with the `vcenter=1`.

Reply:
:+1: I added a quick figure test comparison now.